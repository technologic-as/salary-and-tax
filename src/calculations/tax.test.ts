/* eslint-disable jest/no-conditional-expect */
import {
  calculateDeduction,
  calculateIncomeTax,
  calculateSocialSecurityDeduction,
  calculateStep1,
  calculateStep2,
  calculateStep3,
  calculateStep4,
  getDividendsTaxCalculations,
  getTaxCalculations,
} from './tax';

type TestData = {
  income: number;
  minimumDeduction: string;
  step4: string;
  step2: string;
  step3: string;
  step1: string;
};

const tests: TestData[] = [
  {
    income: -1000,
    minimumDeduction: 'below min',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 0,
    minimumDeduction: 'below min',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 1000,
    minimumDeduction: 'below min',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 3999,
    minimumDeduction: 'below min',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 4000,
    minimumDeduction: 'between min and max',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 5000,
    minimumDeduction: 'between min and max',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 9000,
    minimumDeduction: 'between min and max',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 40000,
    minimumDeduction: 'between min and max',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 80000,
    minimumDeduction: 'between min and max',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 97610,
    minimumDeduction: 'between min and max',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 100000,
    minimumDeduction: 'above max',
    step1: 'below',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 169000,
    minimumDeduction: 'above max',
    step1: 'above',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 170000,
    minimumDeduction: 'above max',
    step1: 'above',
    step2: 'below',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 237900,
    minimumDeduction: 'above max',
    step1: 'above',
    step2: 'above',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 240000,
    minimumDeduction: 'above max',
    step1: 'above',
    step2: 'above',
    step3: 'below',
    step4: 'below',
  },
  {
    income: 598050,
    minimumDeduction: 'above max',
    step1: 'above',
    step2: 'above',
    step3: 'above',
    step4: 'below',
  },
  {
    income: 600000,
    minimumDeduction: 'above max',
    step1: 'above',
    step2: 'above',
    step3: 'above',
    step4: 'below',
  },
  {
    income: 962050,
    minimumDeduction: 'above max',
    step1: 'above',
    step2: 'above',
    step3: 'above',
    step4: 'above',
  },
  {
    income: 999999,
    minimumDeduction: 'above max',
    step1: 'above',
    step2: 'above',
    step3: 'above',
    step4: 'above',
  },
];

const getTestDescription = ({ income, minimumDeduction, step1, step2, step3, step4 }: TestData) => ({
  income: `a income of ${income}`,
  minimumDeduction: `${minimumDeduction} threshold for minimum deduction`,
  step1: `${step1} step one tax`,
  step2: `${step2} step two tax`,
  step3: `${step3} step three tax`,
  step4: `${step4} step four tax`,
});

describe('tax', () => {
  describe('getTaxCalculations should calculate tax', () => {
    tests.map((testData) => {
      const { income, minimumDeduction, step1, step2, step3, step4 } = getTestDescription(testData);
      return it(`on ${income} which is ${minimumDeduction}, ${step1}, ${step2}, ${step3}, ${step4}`, () => {
        expect(getTaxCalculations(testData.income)).toMatchSnapshot();
      });
    });
  });
  describe('calculateDeduction should calculate minimum deduction', () => {
    tests.map((testData) => {
      const { income, minimumDeduction } = getTestDescription(testData);
      return it(`on ${income} which is ${minimumDeduction}`, () => {
        expect(calculateDeduction(testData.income)).toMatchSnapshot();
      });
    });
  });
  describe('calculateIncomeTax should calculate income tax', () => {
    tests.map((testData) => {
      const { income } = getTestDescription(testData);
      return it(`on ${income}`, () => {
        expect(calculateIncomeTax(testData.income)).toMatchSnapshot();
      });
    });
  });
  describe('calculateSocialSecurityDeduction should calculate social security deduction', () => {
    tests.map((testData) => {
      const { income } = getTestDescription(testData);
      return it(`on ${income}`, () => {
        expect(calculateSocialSecurityDeduction(testData.income)).toMatchSnapshot();
      });
    });
  });
  describe('calculateStep1 should calculate step one tax', () => {
    tests.map((testData) => {
      const { income, step1 } = getTestDescription(testData);
      return it(`on ${income} which is ${step1}`, () => {
        const tax = calculateStep1(testData.income);
        expect(tax).toMatchSnapshot();
        if (testData.step1 === 'below') {
          expect(tax).toEqual(0);
        }
      });
    });
  });
  describe('calculateStep2 should calculate step two tax', () => {
    tests.map((testData) => {
      const { income, step2 } = getTestDescription(testData);
      return it(`on ${income} which is ${step2}`, () => {
        const tax = calculateStep2(testData.income);
        expect(tax).toMatchSnapshot();
        if (testData.step2 === 'below') {
          expect(tax).toEqual(0);
        }
      });
    });
  });
  describe('calculateStep3 should calculate step three tax', () => {
    tests.map((testData) => {
      const { income, step3 } = getTestDescription(testData);
      return it(`on ${income} which is ${step3}`, () => {
        const tax = calculateStep3(testData.income);
        expect(tax).toMatchSnapshot();
        if (testData.step3 === 'below') {
          expect(tax).toEqual(0);
        }
      });
    });
  });
  describe('calculateStep4 should calculate step four tax', () => {
    tests.map((testData) => {
      const { income, step4 } = getTestDescription(testData);
      return it(`on ${income} which is ${step4}`, () => {
        const tax = calculateStep4(testData.income);
        expect(tax).toMatchSnapshot();
        if (testData.step4 === 'below') {
          expect(tax).toEqual(0);
        }
      });
    });
  });
  describe('getDividendsTaxCalculations', () => {
    it('should match snapshot', () => {
      expect(getDividendsTaxCalculations(100000)).toMatchSnapshot();
    });
  });
});
