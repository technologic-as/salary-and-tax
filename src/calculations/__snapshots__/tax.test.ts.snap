// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tax calculateDeduction should calculate minimum deduction on a income of -1000 which is below min threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -61550,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 0 which is below min threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -60550,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 1000 which is below min threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -59550,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 3999 which is below min threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -56551,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 4000 which is between min and max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -56550,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 5000 which is between min and max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -55550,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 9000 which is between min and max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -51600,
  "minimumDeduction": 4050,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 40000 which is between min and max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -34550,
  "minimumDeduction": 18000,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 80000 which is between min and max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -12550,
  "minimumDeduction": 36000,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 97610 which is between min and max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -2864,
  "minimumDeduction": 43924,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 100000 which is above max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": -1550,
  "minimumDeduction": 45000,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 169000 which is above max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": 36400,
  "minimumDeduction": 76050,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 170000 which is above max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": 36950,
  "minimumDeduction": 76500,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 237900 which is above max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": 80550,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 240000 which is above max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": 82650,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 598050 which is above max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": 440700,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 600000 which is above max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": 442650,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 962050 which is above max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": 804700,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateDeduction should calculate minimum deduction on a income of 999999 which is above max threshold for minimum deduction 1`] = `
Object {
  "commonIncome": 842649,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
}
`;

exports[`tax calculateIncomeTax should calculate income tax on a income of -1000 1`] = `-220`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 0 1`] = `0`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 1000 1`] = `220`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 3999 1`] = `879`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 4000 1`] = `880`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 5000 1`] = `1100`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 9000 1`] = `1980`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 40000 1`] = `8800`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 80000 1`] = `17600`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 97610 1`] = `21474`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 100000 1`] = `22000`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 169000 1`] = `37180`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 170000 1`] = `37400`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 237900 1`] = `52338`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 240000 1`] = `52800`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 598050 1`] = `131571`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 600000 1`] = `132000`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 962050 1`] = `211651`;

exports[`tax calculateIncomeTax should calculate income tax on a income of 999999 1`] = `219999`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of -1000 1`] = `-82`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 0 1`] = `0`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 1000 1`] = `82`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 3999 1`] = `327`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 4000 1`] = `328`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 5000 1`] = `410`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 9000 1`] = `738`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 40000 1`] = `3280`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 80000 1`] = `6560`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 97610 1`] = `8004`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 100000 1`] = `8199`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 169000 1`] = `13857`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 170000 1`] = `13939`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 237900 1`] = `19507`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 240000 1`] = `19679`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 598050 1`] = `49040`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 600000 1`] = `49200`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 962050 1`] = `78888`;

exports[`tax calculateSocialSecurityDeduction should calculate social security deduction on a income of 999999 1`] = `81999`;

exports[`tax calculateStep1 should calculate step one tax on a income of -1000 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 0 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 1000 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 3999 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 4000 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 5000 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 9000 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 40000 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 80000 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 97610 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 100000 which is below step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 169000 which is above step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 170000 which is above step one tax 1`] = `0`;

exports[`tax calculateStep1 should calculate step one tax on a income of 237900 which is above step one tax 1`] = `1204`;

exports[`tax calculateStep1 should calculate step one tax on a income of 240000 which is above step one tax 1`] = `1244`;

exports[`tax calculateStep1 should calculate step one tax on a income of 598050 which is above step one tax 1`] = `1351`;

exports[`tax calculateStep1 should calculate step one tax on a income of 600000 which is above step one tax 1`] = `1351`;

exports[`tax calculateStep1 should calculate step one tax on a income of 962050 which is above step one tax 1`] = `1351`;

exports[`tax calculateStep1 should calculate step one tax on a income of 999999 which is above step one tax 1`] = `1351`;

exports[`tax calculateStep2 should calculate step two tax on a income of -1000 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 0 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 1000 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 3999 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 4000 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 5000 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 9000 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 40000 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 80000 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 97610 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 100000 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 169000 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 170000 which is below step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 237900 which is above step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 240000 which is above step two tax 1`] = `0`;

exports[`tax calculateStep2 should calculate step two tax on a income of 598050 which is above step two tax 1`] = `14800`;

exports[`tax calculateStep2 should calculate step two tax on a income of 600000 which is above step two tax 1`] = `14882`;

exports[`tax calculateStep2 should calculate step two tax on a income of 962050 which is above step two tax 1`] = `15617`;

exports[`tax calculateStep2 should calculate step two tax on a income of 999999 which is above step two tax 1`] = `15617`;

exports[`tax calculateStep3 should calculate step three tax on a income of -1000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 0 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 1000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 3999 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 4000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 5000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 9000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 40000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 80000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 97610 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 100000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 169000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 170000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 237900 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 240000 which is below step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 598050 which is above step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 600000 which is above step three tax 1`] = `0`;

exports[`tax calculateStep3 should calculate step three tax on a income of 962050 which is above step three tax 1`] = `45480`;

exports[`tax calculateStep3 should calculate step three tax on a income of 999999 which is above step three tax 1`] = `45843`;

exports[`tax calculateStep4 should calculate step four tax on a income of -1000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 0 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 1000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 3999 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 4000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 5000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 9000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 40000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 80000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 97610 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 100000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 169000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 170000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 237900 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 240000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 598050 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 600000 which is below step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 962050 which is above step four tax 1`] = `0`;

exports[`tax calculateStep4 should calculate step four tax on a income of 999999 which is above step four tax 1`] = `5702`;

exports[`tax getDividendsTaxCalculations should match snapshot 1`] = `
Object {
  "afterDividendsTax": 53290,
  "dividends": 78000,
  "dividendsTax": 24710,
  "surplus": 100000,
  "surplusTax": 22000,
  "upwardsAdjustment": 112320,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of -1000 which is below min threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 12623,
  "commonIncome": -61550,
  "income": -1000,
  "incomeTax": -13541,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
  "socialSecurityDeduction": -82,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": -13623,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 0 which is below min threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 13321,
  "commonIncome": -60550,
  "income": 0,
  "incomeTax": -13321,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 0,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": -13321,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 1000 which is below min threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 14019,
  "commonIncome": -59550,
  "income": 1000,
  "incomeTax": -13101,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 82,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": -13019,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 3999 which is below min threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 16114,
  "commonIncome": -56551,
  "income": 3999,
  "incomeTax": -12442,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 327,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": -12115,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 4000 which is between min and max threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 16113,
  "commonIncome": -56550,
  "income": 4000,
  "incomeTax": -12441,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 328,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": -12113,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 5000 which is between min and max threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 16811,
  "commonIncome": -55550,
  "income": 5000,
  "incomeTax": -12221,
  "minimumDeduction": 4000,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 410,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": -11811,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 9000 which is between min and max threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 19614,
  "commonIncome": -51600,
  "income": 9000,
  "incomeTax": -11352,
  "minimumDeduction": 4050,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 738,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": -10614,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 40000 which is between min and max threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 44321,
  "commonIncome": -34550,
  "income": 40000,
  "incomeTax": -7601,
  "minimumDeduction": 18000,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 3280,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": -4321,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 80000 which is between min and max threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 76201,
  "commonIncome": -12550,
  "income": 80000,
  "incomeTax": -2761,
  "minimumDeduction": 36000,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 6560,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": 3799,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 97610 which is between min and max threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 90237,
  "commonIncome": -2864,
  "income": 97610,
  "incomeTax": -631,
  "minimumDeduction": 43924,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 8004,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": 7373,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 100000 which is above max threshold for minimum deduction, below step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 92142,
  "commonIncome": -1550,
  "income": 100000,
  "incomeTax": -341,
  "minimumDeduction": 45000,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 8199,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": 7858,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 169000 which is above max threshold for minimum deduction, above step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 147135,
  "commonIncome": 36400,
  "income": 169000,
  "incomeTax": 8008,
  "minimumDeduction": 76050,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 13857,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": 21865,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 170000 which is above max threshold for minimum deduction, above step one tax, below step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 147932,
  "commonIncome": 36950,
  "income": 170000,
  "incomeTax": 8129,
  "minimumDeduction": 76500,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 13939,
  "step1": 0,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": 22068,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 237900 which is above max threshold for minimum deduction, above step one tax, above step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 199468,
  "commonIncome": 80550,
  "income": 237900,
  "incomeTax": 17721,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 19507,
  "step1": 1204,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": 38432,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 240000 which is above max threshold for minimum deduction, above step one tax, above step two tax, below step three tax, below step four tax 1`] = `
Object {
  "afterTax": 200894,
  "commonIncome": 82650,
  "income": 240000,
  "incomeTax": 18183,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 19679,
  "step1": 1244,
  "step2": 0,
  "step3": 0,
  "step4": 0,
  "totalTax": 39106,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 598050 which is above max threshold for minimum deduction, above step one tax, above step two tax, above step three tax, below step four tax 1`] = `
Object {
  "afterTax": 435905,
  "commonIncome": 440700,
  "income": 598050,
  "incomeTax": 96954,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 49040,
  "step1": 1351,
  "step2": 14800,
  "step3": 0,
  "step4": 0,
  "totalTax": 162145,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 600000 which is above max threshold for minimum deduction, above step one tax, above step two tax, above step three tax, below step four tax 1`] = `
Object {
  "afterTax": 437184,
  "commonIncome": 442650,
  "income": 600000,
  "incomeTax": 97383,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 49200,
  "step1": 1351,
  "step2": 14882,
  "step3": 0,
  "step4": 0,
  "totalTax": 162816,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 962050 which is above max threshold for minimum deduction, above step one tax, above step two tax, above step three tax, above step four tax 1`] = `
Object {
  "afterTax": 643680,
  "commonIncome": 804700,
  "income": 962050,
  "incomeTax": 177034,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 78888,
  "step1": 1351,
  "step2": 15617,
  "step3": 45480,
  "step4": 0,
  "totalTax": 318370,
}
`;

exports[`tax getTaxCalculations should calculate tax on a income of 999999 which is above max threshold for minimum deduction, above step one tax, above step two tax, above step three tax, above step four tax 1`] = `
Object {
  "afterTax": 664105,
  "commonIncome": 842649,
  "income": 999999,
  "incomeTax": 185382,
  "minimumDeduction": 100800,
  "personalAllowance": 56550,
  "socialSecurityDeduction": 81999,
  "step1": 1351,
  "step2": 15617,
  "step3": 45843,
  "step4": 5702,
  "totalTax": 335894,
}
`;
